### ISSUE TITLE
Flutter web-renderer canvas and Myanmar font

### ISSUE THREAD

**Issue by altotunchitoo** [flutter/flutter#88150]
<img width="520" alt="Screen Shot 2021-08-13 at 13 09 19" src="https://user-images.githubusercontent.com/51152835/129315494-500c8ab0-100b-43b4-a1f1-befcf09f9c50.png">

In the screenshot, you can see how it's incorrect in rendering Myanmar font.
It's still correct "ထမင်းမစားမီဆိုအပ်သော" but "မေတ္တာ" is incorrect.
The case comes from "္" (0xE1 0x80 0xB9).

**Comment by darshankawar** [comment]
@altotunchitoo 
Please provide `flutter doctor -v`, a [complete minimal code sample](https://stackoverflow.com/help/minimal-reproducible-example) in properly formatted text (no screenshots).

**Comment by altotunchitoo** [comment]
# flutter doctor -v
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.2 20G95 darwin-arm, locale en-MM)
    • Flutter version 2.2.3 at /Users/altotunchitoo/Development/code/flutter
    • Framework revision f4abaa0735 (6 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/altotunchitoo/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    ✗ Android SDK file not found:
      /Users/altotunchitoo/Library/Android/sdk/platforms/android-31/android.jar.
    • Try re-installing or updating your Android SDK,
      visit https://flutter.dev/docs/get-started/install/macos#android-setup for
      detailed instructions.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 59.0.4
    • Dart plugin version 212.4746.57

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (3 available)
    • iPhone 12 Pro (mobile) • B4A34C4A-087B-4182-AE2A-8733F615DB07 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   •
      macOS 11.5.2 20G95 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript •
      Google Chrome 92.0.4515.131

! Doctor found issues in 2 categories.

# Minimal code sample

import 'package:flutter/material.dart';

class MyanmarText extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(child: Text("မြန်မာစာ ပါဌ်ဆင့်ဥပမာတစ်ခုမှာ \"မင်္ဂလာပါ\"")));
  }
}

# Notice
The result should be displayed မြန်မာစာ  ပါဌ်ဆင့်ဥပမာတစ်ခုမှာ "မင်္ဂလာပါ".
Myanmar font which contains "ပါဌ်ဆင့်" (I don't know how to call it in English :P)  is not correct in web rendering with canvas.

**Comment by ygnCybernoob** [comment]
Hi  @darshankawar,  I just found **Flutter encoded ** the **Myanmar text**  for canvas  in **Zawgyi** Charset which is not standard. 

```dart 
Text('မင်္ဂလာပါ'); //Mingalarpar is just greeting :) 
```

then, build for Flutter web in cavaskit renderer and I searched for charcode in main.js and I found this
`\u1019\u1004\u103a\u1039\u1002\u101c\u102c\u1015\u102b` which is encoded in Zawgyi Charset

This is how Flutter should produced in **Unicode standard** `\u1019\u1004\u103a\u1039\u1002\u101c\u102c\u1015\u102b` 


For more info for **Myanmar Unicode**: [Myanmar Unicode Block](https://en.m.wikipedia.org/wiki/Myanmar_(Unicode_block))

**Comment by darshankawar** [comment]
@altotunchitoo 
I ran your code sample as-is using latest master with canvaskit rendering and saw below result:

<img width="296" alt="Screenshot 2021-08-16 at 12 23 42 PM" src="https://user-images.githubusercontent.com/67046386/129523127-1215fa87-49cb-4eec-884d-0e888b499238.png">

which looks like your expected result.

```
class MyanmarText extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(child: Text("မြန်မာစာ ပါဌ်ဆင့်ဥပမာတစ်ခုမှာ မင်္ဂလာပါ")));
  }
}
```

<details>
<summary> master flutter doctor -v </summary>

```
[✓] Flutter (Channel master, 2.5.0-7.0.pre.83, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.5.0-7.0.pre.83 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 06be7e5392 (12 hours ago), 2021-08-15 14:37:02 -0400
    • Engine revision 1af0a20793
    • Dart version 2.15.0 (build 2.15.0-15.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • SM G975F (mobile) • RZ8M802WY0X • android-arm64 • Android 10 (API 29)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 91.0.4472.114



```

</details>

Can you check above to see if it's correct interpretation of your issue ?

**Comment by altotunchitoo** [comment]
Humm,
@darshankawar  I don't know what's wrong with me.

# This is the screen record.
https://user-images.githubusercontent.com/51152835/129525157-f5547815-659e-4215-8799-009258c4a55b.mov

# This is the screenshot.
[https://test-mm-font.alto.work](https://test-mm-font.alto.work)
It's not correct on PC's browser  but as an exception the test site's Myanmar font is correct on Mobile devices.
The sentence is written in Unicode standard format.
<img width="1552" alt="Screen Shot 2021-08-16 at 13 47 58" src="https://user-images.githubusercontent.com/51152835/129525765-a1b83203-b11e-4531-bd25-903af0c79e28.png">

**Comment by altotunchitoo** [comment]
@darshankawar 
I noticed that I've used stable channel and you're using master channel.
The case is solved **by switching channel from stable 2.2.3 to master 2.5.0-7.0.pre.83**.
Thank you very much!
